### ISSUE TITLE
Kubernetes UI doesn't start correctly with k8s 1.6

### ISSUE THREAD

**Issue by galal-hussein** [rancher/rancher#9127]
**Rancher versions:**
rancher/server: v1.6.1-rc3
kubernetes (if applicable): rancher/k8s:v1.6.6-rancher1-4

**Steps to Reproduce:**

- Install kubernetes 1.6.6 from k8s1.6-dev

**Results:**

The kubernetes stack deployed successfully and i can see the addon pods started correctly, however when hovering over the kubernetes UI, i am still seeing setting up, also i am not able to access kubectl UI. 

![k8s](https://user-images.githubusercontent.com/3309968/27402166-606b198a-56ce-11e7-9436-7a43bf2210f0.png)

Rancher server logs:

```
time="2017-06-21T18:52:08Z" level=info msg="pulling rancher/etcd:v2.3.7-11 image." service=gms 
time="2017-06-21T18:52:08Z" level=info msg="pulling rancher/healthcheck:v0.3.1 image." service=gms 
time="2017-06-21T18:52:08Z" level=info msg="pulling rancher/network-manager:v0.7.1 image." service=gms 
time="2017-06-21T18:52:08Z" level=info msg="pulling rancher/kubectld:v0.6.6 image." service=gms 
time="2017-06-21T18:52:08Z" level=info msg="pulling rancher/metadata:v0.9.1 image." service=gms 
time="2017-06-21T18:52:08Z" level=info msg="pulling rancher/dns:v0.15.1 image." service=gms 
time="2017-06-21T18:52:08Z" level=info msg="pulling rancher/net:holder image." service=gms 
time="2017-06-21T18:52:08Z" level=info msg="pulling rancher/net:v0.11.2 image." service=gms 
time="2017-06-21T18:52:08Z" level=info msg="pulling rancher/net:v0.11.2 image." service=gms 
time="2017-06-21T18:52:08Z" level=info msg="pulling rancher/k8s:v1.6.6-rancher1-4 image." service=gms 
time="2017-06-21T18:52:08Z" level=info msg="pulling rancher/etc-host-updater:v0.0.2 image." service=gms 
time="2017-06-21T18:52:08Z" level=info msg="pulling rancher/k8s:v1.6.6-rancher1-4 image." service=gms 
time="2017-06-21T18:52:08Z" level=info msg="pulling rancher/k8s:v1.6.6-rancher1-4 image." service=gms 
time="2017-06-21T18:52:08Z" level=info msg="pulling rancher/k8s:v1.6.6-rancher1-4 image." service=gms 
time="2017-06-21T18:52:08Z" level=info msg="pulling rancher/kubernetes-agent:v0.6.2 image." service=gms 
time="2017-06-21T18:52:08Z" level=info msg="pulling rancher/k8s:v1.6.6-rancher1-4 image." service=gms 
time="2017-06-21T18:52:08Z" level=info msg="pulling rancher/lb-service-rancher:v0.7.4 image." service=gms 
time="2017-06-21T18:52:08Z" level=info msg="pulling rancher/k8s:v1.6.6-rancher1-4 image." service=gms 
time="2017-06-21T18:52:26Z" level=error msg="Error looking up token." error="HTTP error: 404 Not Found, 404" 
time="2017-06-21T18:53:11Z" level=error msg="Error looking up token." error="HTTP error: 404 Not Found, 404" 
time="2017-06-21T18:53:12Z" level=error msg="Error looking up token." error="HTTP error: 404 Not Found, 404" 
time="2017-06-21T18:53:37Z" level=error msg="Error looking up token." error="HTTP error: 404 Not Found, 404" 
time="2017-06-21T18:53:52Z" level=error msg="Error looking up token." error="HTTP error: 404 Not Found, 404" 
time="2017-06-21T18:53:54Z" level=error msg="Error looking up token." error="HTTP error: 404 Not Found, 404" 
time="2017-06-21T18:53:56Z" level=error msg="Error looking up token." error="HTTP error: 404 Not Found, 404" 
time="2017-06-21T18:54:28Z" level=error msg="Error looking up token." error="HTTP error: 404 Not Found, 404" 
time="2017-06-21T18:54:54Z" level=error msg="Error looking up token." error="HTTP error: 404 Not Found, 404" 
time="2017-06-21T18:54:54Z" level=error msg="Error looking up token." error="HTTP error: 404 Not Found, 404" 
time="2017-06-21T18:54:58Z" level=error msg="Error looking up token." error="HTTP error: 404 Not Found, 404" 
time="2017-06-21T18:54:59Z" level=error msg="Error looking up token." error="HTTP error: 404 Not Found, 404" 
time="2017-06-21T18:55:00Z" level=error msg="Error looking up token." error="HTTP error: 404 Not Found, 404" 
time="2017-06-21T18:55:01Z" level=error msg="Error looking up token." error="HTTP error: 404 Not Found, 404" 
time="2017-06-21T18:55:05Z" level=error msg="Error looking up token." error="HTTP error: 404 Not Found, 404" 
```

**Comment by deniseschannon** [comment]
This is related to having an unauthenticated Rancher setup.

**Comment by deniseschannon** [comment]
This is available in master and v1.6 branch of cattle.

**Comment by galal-hussein** [comment]
Rancher server: v1.6.3-rc2
k8s image: rancher/k8s:v1.6.6-rancher1-4

Steps to validate:

- Install Rancher server
- Create new k8s environment
- Register agents

I don't see the problem anymore, and i can see Kubectl UI and dahsboard.
